New article: TWVycnkgQ2hyaXN0bWFzIDop

thibaultduponchelle opened the pull request:
Hello,

Here is another little story about elves, Perl and MIME:base64 🎅🏼

thibaultduponchelle commented:
@vanHoesel Would you be available to review this article? Please 🙏🏼

thibaultduponchelle commented:
Hi @adamcrussell

As discussed, if you could help editing and correcting this article, it would be a great 👍🏼

adamcrussell reviewed (approved):
I like this! I especially like the style and backstory. I believe the Advent Calendar entries should have a distinct holiday theme like this. 

1. The spelling sirup is much less common than syrup. No need to change it, in my opinion.
2. I would add maybe a little more content. Another example or two. This could be fine as is, but if so inspired I think the typical Advent Calendar entry is a little longer.

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,66 @@
+
+=head2 Perl birthday party
+
+On 18th of December, to celebrate the Perl birthday, elves organized a big party with cakes, waffles, pancakes and maple sirup.
```
> 
```suggestion
On the 18th of December, to celebrate Perl's birthday, the elves organized a big party with cakes, waffles, pancakes and maple sirup.
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,66 @@
+
+On 18th of December, to celebrate the Perl birthday, elves organized a big party with cakes, waffles, pancakes and maple sirup.
+
+They party like crazy but only until exactly midnight where everyone went to bed.
```
```suggestion
They partied like crazy but only until exactly midnight when everyone went to bed.
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,66 @@
+
+They party like crazy but only until exactly midnight where everyone went to bed.
+
+The day after, elves walked to North Pole warehouse. To open the door and disable the alarm, as every day, they called the dedicated elf "spof" to enter the code.
```
```suggestion
The day after, the elves walked to North Pole warehouse. To open the door and disable the alarm, as every day, they called the dedicated elf "spof" to enter the code.
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,66 @@
+
+- Type it again, said another elf
+
+He proceed and got once again "ERROR"
```
```suggestion
He proceeded and, once again, got "ERROR".
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,66 @@
+
+=begin perl
+
+	use MIME::Base64;
```
```suggestion
	use MIME::Base64 qw( decode_base64 );
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,66 @@
+=begin perl
+
+	use Env;
+	use MIME::Base64;
```
```suggestion
	use MIME::Base64 qw( encode_base64 );
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,66 @@
+
+=head2 Conclusion
+
+You should never feed a L<Gremlin|https://en.wikipedia.org/wiki/Gremlins> after midnight but you also need to monitor closely the maple sirup consumption of elves!
```
```suggestion
You should never feed L<Gremlin|https://en.wikipedia.org/wiki/Gremlins> after midnight but you also need to monitor closely the maple sirup consumption of elves!
```

oalders reviewed (changes requested):
Love it! Just a few minor things. 🧝🏻

thibaultduponchelle commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,66 @@
+
+=head2 Deciphering the code
+
+One of the elves has little experience with IT and he immediately thought he would test L<MIME::Base64> to translate this gibberish:
```
Yes, good catch thx

thibaultduponchelle commented:
I added a paragraph upon remark of @adamcrussell and fixed all requested changes. 

Except edits in the new paragraph, I think it could be ready to merge.

thibaultduponchelle commented:
```
$ perl t/article_pod.t 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod 
# Subtest: 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
    # Subtest: header for 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
        ok 1 - Required field <title> is there
        ok 2 - Required field <topic> is there
        ok 3 - Required field <author> is there
        1..3
    ok 1 - header for 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
    # Subtest: pod for 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
        ok 1 - Pod checks passes
        1..1
    ok 2 - pod for 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
    1..2
ok 1 - 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod
1..1
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,86 @@
+
+=end perl
+
+The above script is using environment variable to pass the secret.
```
```suggestion
The above script is using an environment variable to pass the secret.
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,86 @@
+
+=head2 Make perl speak Spof's dialect
+
+Aside mapple syrup, Spof enjoys doing some Perl scripting for himself the evening, so after this event he did (or at least he I<tried>).
```
```suggestion
Aside from maple syrup, Spof enjoys doing some Perl scripting for himself the evening, so after this event he did (or at least he I<tried>).
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,86 @@
+
+Being still confused he tried to write C<cHJpbnQgIk1lcnJ5IENocmlzdG1hcyA6KSIK> but it was not working.
+
+Confused but lazy, he dig the L<MetaCPAN|https://metacpan.org/> for a solution and went with L<Acme::Base64>:
```
```suggestion
Confused but lazy, searched the L<MetaCPAN|https://metacpan.org/> for a solution and went with L<Acme::Base64>:
```

oalders commented on 2022/incoming/TWVycnkgQ2hyaXN0bWFzIDop.pod:
```diff
@@ -0,0 +1,86 @@
+
+=end perl
+
+This way, C<perl> interpreter could understand the B<base64> dialect spoken by Spof! Magic!
```
```suggestion
This way, the C<perl> interpreter could understand the B<base64> dialect spoken by Spof! Magic!
```

oalders reviewed (changes requested):
Great stuff! Just minor suggestions.

thibaultduponchelle commented:
@oalders I updated, ready to merge?

oalders commented:
@thibaultduponchelle, sure. Could you squash it down to a single commit?

thibaultduponchelle commented:
Done

oalders reviewed (approved):
\o/

oalders commented:
I guess we can also squash via the merge button. I had forgotten about that.
